include/generator

```diff
@@ -227,26 +227,37 @@ class __manual_lifetime<_T&&> {
     _T* __value_;
 };
 
+struct use_allocator_arg {};
+
 namespace ranges {
 
-template <typename _Rng>
+template <typename _Rng, typename _Allocator = use_allocator_arg>
```

Comment:
I don't think `use_allocator_arg` is the right default here.
It's not an allocator and so can't be used as an allocator (as you are using it below).
Perhaps the default of `std::allocator<void>` is more appropriate?